### PR TITLE
Verify contribution form filled label

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28759

## Description
Added status check to verify that `Contribution Form Filled` label exists on external PRs.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: .github/workflows/check-contribution-form-filled.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          python -m pip install --upgrade pip
+          pip install pipenv
+      - name: Check if contribution form needs to be filled
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo "Checking if contribution form needs to be filled for PR: $PR_NUMBER"
+          cd Utils/github_workflow_scripts
+          pipenv install --dev
```
Comment (ShahafBenYakir) [inline .github/workflows/check-contribution-form-filled.yml:35]: Why this isn't part of the Install Python Dependencies step

File: .github/workflows/check-contribution-form-filled.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          ref: master
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Print Context
+        run: |
+          echo "$GITHUB_CONTEXT"
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
```
Comment (ShahafBenYakir) [inline .github/workflows/check-contribution-form-filled.yml:23]: No need for this

File: .github/workflows/check-contribution-form-filled.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    if: github.repository == 'demisto/content' && github.event.pull_request.head.repo.fork == true && contains(github.head_ref, 'xsoar-bot-contrib-ContributionTestPack') == false && contains(github.event.pull_request.labels.*.name, 'Contribution Form Filled') == false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: master
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
```
Comment (ShahafBenYakir) [inline .github/workflows/check-contribution-form-filled.yml:18]: why do we need this?

File: Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    Args:
+        pr_files (PaginatedList[File]): The list of pr files
+
+    Returns:
+        The pack name
+
+    """
+    for file in pr_files:
+        if PACKS in file.filename:
+            return re.findall(PACK_NAME_REGEX, file.filename)[0]
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py:80]: What if there are two different pack names in the PR? We should take both

File: Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    github_client: Github = Github(github_token, verify=False)
+    content_repo: Repository = github_client.get_repo(f'{org_name}/{repo_name}')
+    pr: PullRequest = content_repo.get_pull(int(pr_number))
+    pr_files = pr.get_files()
+    pack_name: str = get_pack_name_from_pr(pr_files)
+
+    if pr_metadata_filename := get_metadata_filename_from_pr(pr_files):
+        support_type = get_pack_support_type_from_pr_metadata_file(pr_metadata_filename, pr)
+    else:
+        support_type = get_pack_support_type_from_repo_metadata_file(pack_name)
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py:131]:
```suggestion 

    if pr_metadata_filename := get_metadata_filename_from_pr(pr_files):
        support_type = get_pack_support_type_from_pr_metadata_file(pr_metadata_filename, pr)
    else:
        pack_name: str = get_pack_name_from_pr(pr_files)
        support_type = get_pack_support_type_from_repo_metadata_file(pack_name)
```

File: Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    """ Iterates over all pr files and return the pr metadata.json filename if exists, else None
+
+    Args:
+        pr_files (PaginatedList[File]): The list of pr files
+
+    Returns:
+        The pr metadata.json filename if exists, else None
+
+    """
+    print(f'Looking for a {METADATA} file in the PR.')
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py:36]:
```suggestion
    print(Searching for a {METADATA} file in the PR.')
```

File: Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        The support type
+
+    """
+    print(f'Getting support type from {pr_metadata_filename}.')
+    _, branch_name = pr.head.label.split(':')
+    print(f'Branch name is: {branch_name}')
+    contributor_repo: Repository = pr.head.repo
+    branch: Branch = contributor_repo.get_branch(branch=branch_name)
+    metadata_file: ContentFile = contributor_repo.get_contents(path=pr_metadata_filename, ref=branch.commit.sha)
+    metadata_file_content: dict = json.loads(base64.b64decode(metadata_file.content))
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py:64]: Does metadata_file.content containes encoded data?

File: Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+    org_name: str = 'demisto'
+    repo_name: str = 'content'
+    github_client: Github = Github(github_token, verify=False)
+    content_repo: Repository = github_client.get_repo(f'{org_name}/{repo_name}')
+    pr: PullRequest = content_repo.get_pull(int(pr_number))
+    pr_files = pr.get_files()
+    pack_name: str = get_pack_name_from_pr(pr_files)
+
+    if pr_metadata_filename := get_metadata_filename_from_pr(pr_files):
```
Comment (ShahafBenYakir) [inline Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py:128]: Need to consider the case that there is more than one pack in the PR. The code should iterate on all and check for all support types, if there is at least one that is not XSOAR then fail.

**Review by ShahafBenYakir — COMMENTED** [review]
Nice.
Let's add a test for the `load_json` function

File: Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py (hunk 7 above)
Comment (guyfreund) [reply Utils/github_workflow_scripts/check_if_needs_to_fill_contribution_form.py:64]: yes

File: .github/workflows/check-contribution-form-filled.yml (hunk 3 above)
Comment (guyfreund) [reply .github/workflows/check-contribution-form-filled.yml:18]: for the pipenv

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**